eth/backend.go

```diff
@@ -549,16 +552,23 @@ func New(ctx context.Context, stack *node.Node, config *ethconfig.Config, logger
 	}
 
 	var heimdallClient heimdall.HeimdallClient
+	var polygonBridge *bridge.Bridge
 
 	if chainConfig.Bor != nil {
 		if !config.WithoutHeimdall {
 			heimdallClient = heimdall.NewHeimdallClient(config.HeimdallURL, logger)
 		}
 
+		if config.PolygonSync || config.PolygonSyncStage {
+			bridgeDB := polygoncommon.NewDatabase(config.Dirs.DataDir, logger)
+			bridgeStore := bridge.NewStore(bridgeDB)
+			polygonBridge = bridge.NewBridge(bridgeStore, logger, consensusConfig.(*borcfg.BorConfig), heimdallClient.FetchStateSyncEvents, bor.GenesisContractStateReceiverABI())
```

Comment:
One suggestion I have is: we can wrap these 3 init lines in a `bridge.Assemble(logger, datadir, borCfg, heimdallClient) *Bridge` function to encapsulate the logic of assembling the bridge inside the `bridge` package. Then here we simply can call
```
if polygonSync || polygonSyncStage {
    polygonBridge = bridge.Assemble()
}
```